synapse/storage/databases/main/events_worker.py

```diff
@@ -491,9 +492,12 @@ async def _get_events_from_cache_or_db(self, event_ids, allow_rejected=False):
         return event_entry_map
 
     def _invalidate_get_event_cache(self, event_id):
+        if self._external_cache.is_enabled():
+            self._external_cache.delete("getEvent", event_id)
```

Comment:
you have to `await` a coroutine for it to actually do anything.

Reply:
I've since made this a background process, as we depend on this function to be sync all over the place. I am a bit worried that doing this might break areas of Synapse that rely on this cache to be accurate though?

Reply:
Following on from https://github.com/matrix-org/synapse/pull/9379#issuecomment-777386106, we should probably use two caches.